### PR TITLE
Add PrivateAPIServer to the public API

### PR DESCRIPTION
**Pull request by asalkeld**
```release-note
Add PrivateAPIServer field to MasterPoolProfile 
```

### CODE REVIEW

File: pkg/api/2019-10-27-preview/convertertointernal.go [hunk 1]
```diff
@@ -90,6 +90,9 @@ func mergeProperties(oc *OpenShiftManagedCluster, cs *api.OpenShiftManagedCluste
 	if oc.Properties.FQDN != nil {
 		cs.Properties.FQDN = *oc.Properties.FQDN
 	}
+	if oc.Properties.MasterPoolProfile != nil && oc.Properties.MasterPoolProfile.PrivateAPIServer != nil {
+		cs.Properties.PrivateAPIServer = *oc.Properties.MasterPoolProfile.PrivateAPIServer
+	}
```
Comment (ehashman) [inline pkg/api/2019-10-27-preview/convertertointernal.go]: Is this something that should be updateable? What would the user expect to happen if this was switched from "true" to "false" or vice versa?

File: pkg/cluster/hash_test.go [hunk 2]
```diff
@@ -155,6 +155,7 @@ func TestHashScaleSetStability(t *testing.T) {
 			cs.Config.Images.ImagePullSecret = []byte{}
 			cs.Config.Images.GenevaImagePullSecret = []byte{}
 		default:
+			cs.Properties.NetworkProfile.ManagementSubnetCIDR = nil
```
Comment (ehashman) [inline pkg/cluster/hash_test.go]: Why do we need to set this in this PR?

Comment (asalkeld) [reply pkg/cluster/hash_test.go]: removed

**Comment by asalkeld** [comment]
flake https://github.com/openshift/openshift-azure/issues/1775

**Pull request merged**